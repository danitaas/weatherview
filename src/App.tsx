import * as React from "react";
import logo from "./logo.svg";
import Location from "./components/Location/Location";
import Weather from "./components/Weather/Weather";
import "./App.css";

class App extends React.Component {
    public render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <h1 className="App-title">Welcome to Weather Viewer</h1>
                </header>
                <div>
                    <Location />
                    <Weather />
                </div>
            </div>
        );
    }
}

export default App;
