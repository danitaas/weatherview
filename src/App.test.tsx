import * as React from "react";
import App from "./App";
// todo: cant use typescript import without correcting toolchain
const ReactShallowRenderer = require("react-test-renderer/shallow");
//const ReactTestRenderer = require("react-test-renderer");

it("renders shallow without crashing", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<App />);
    const result = renderer.getRenderOutput();
    expect(result.type).toBe("div");
});

it("renders snapshot", () => {
    const renderer = new ReactShallowRenderer();
    const result = renderer.render(<App />);
    expect(result).toMatchSnapshot();
});
