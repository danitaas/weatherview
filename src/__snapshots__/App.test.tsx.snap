// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src={undefined}
    />
    <h1
      className="App-title"
    >
      Welcome to Weather Viewer
    </h1>
  </header>
  <div
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <Connect(Location) />
    <Connect(Weather) />
  </div>
</div>
`;
